### PR TITLE
Removing httprouter

### PR DESCRIPTION
**Pull request by jprobinson**
We've found including this dependency has continued to be more of a pain than a benefit.

I know of no teams currently using this router, it has made breaking changes since the last tagged release and, from looking at the issues and PRs, it appears to longer be maintained.

### CODE REVIEW

File: router.go [hunk 1]
```diff
@@ -25,16 +24,13 @@ type RouterOption func(Router) Router
...
-// * `httprouter` which uses github.com/julienschmidt/httprouter
 //
 // If the supplied name does not match any known router, `gorilla` will be used.
 // If a user wishes to supply their own router implementation, the `CustomRouter` option
 // is available.
 func RouterSelect(name string) RouterOption {
 	return func(_ Router) Router {
 		switch name {
-		case "httprouter":
-			return &FastRouter{httprouter.New()}
```
Comment (oliver-456) [inline router.go:37]: Do this cold turkey or give users a warning that this has been deprecated and now uses GorillaRouter?

**Review by oliver-456 — APPROVED** [review]
👍 for doing this, that'll reduce our dep-pinning mess by 2oo%.

Just one comment regarding the deprecation and if you want to issue a warning for a bit.

File: router.go (hunk 1 above)
Comment (jprobinson) [reply router.go:37]: We dont have a context at this point so any log message would get lost into nowhere land :(

I think we're fine cold turkey, though. I've been surveying users and no one has piped up in defense of httprouter.

If all else fails, the repo is already tagged so those poor quiet souls could pin the repo until they move to the "slower" router impls.

Comment (oliver-456) [reply router.go:37]: Ah, good point, you don't have `ctx` here yet.

🚢 it

**Pull request merged**